options/options_test.go

```diff
@@ -25,6 +27,55 @@ const (
 	ShouldFail
 )
 
+// TOOLS-2287 specifies the desired behavior in the case that
+// the user specifies unsupported URI options.
+// This test case ensures the proper logging behavior.
+func TestLogUnsupportedOptions(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.UnitTestType)
+	
+	type TestCase struct {
+		description              string
+		args, unsupportedOptions []string
+	}
+
+	// Test matrix for logging behavior
+	cases := []TestCase{
```

Comment:
I'm familiar with the structure you're suggesting, but I figured it'd be a good idea to do it as a test matrix in case we want to drop more of our code in favor of driver functionality sometime in the future.

When/if we do this, we can easily expand this test case under the current structure. I'm amenable to the change, if you think it's necessary, but are you certain that this wouldn't be useful sometime in the future?

Reply:
> but are you certain that this wouldn't be useful sometime in the future?
This is a trap that's easy to fall into, and the rule of thumb is "YAGNI". I would say it's rather unlikely, that we will need to add any more test cases that match this particular format.

If the situation does arise in the future where this test needs to be expanded, it's better to leave it to the implementer to refactor according to their specific needs, as opposed to implementing a factoring that we don't really need right now based on assumptions about what someone might need in the future.

Reply:
Sounds good! I will apply the recommended changes!